source/js/components/search-form.js

```diff
@@ -0,0 +1,108 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content');
+    const searchFieldSet = document.querySelector('.js-search-fieldset');
+    let searchInputEl;
+    const searchButtonEl = formEl.querySelector('.js-search-btn');
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        searchInputEl = formEl.querySelector('.js-search');
+        formEl.addEventListener('submit', searchSubmit);
+    }
+
+    function searchSubmit(e) {
+        e.preventDefault();
+        getData();
+    }
+
+    function getData() {
+        preloader(true);
+        fetchUtils.getJSON(
+            url,
+            {
+                body: JSON.stringify(searchInputEl.value),
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }
+        )
+        .then((data) => {
+            preloader(false);
+            renderResults(data.results);
```

Comment:
[tip] In a real-world application, it's important to make sure you understand what the API contract looks like, and what happens when data is missing. For example, you're making an assumption here that the `results` node of the returned data exists, and then in  `renderDoctorsSearchResult` you're assuming it's an array and looping through it. This probably just means having a clear conversation with your back-end engineer and documenting the API contract, but if there's any chance of data variations, be sure to account for them in your code.